File: tekhook-global-sms-quickstart/node/tekhook.js
// tekhook.js — Lightweight Tekhook API wrapper
// Full API docs: https://tekhook.co

const BASE_URL = 'https://api.tekhook.co/v1';

class SMSClient {
  constructor(apiKey) {
    this.apiKey = apiKey;
  }

  async send({ to, message, from }) {
    const res = await fetch(`${BASE_URL}/sms/send`, {
      method: 'POST',
      headers: {
        Authorization: `Bearer ${this.apiKey}`,
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({ to, message, from }),
    });

    if (!res.ok) {
      const err = await res.json();
      throw new Error(`Tekhook API error: ${err.message || res.statusText}`);
    }

    return res.json();
  }

  async sendBulk({ recipients, message, from }) {
    const res = await fetch(`${BASE_URL}/sms/bulk`, {
      method: 'POST',
      headers: {
        Authorization: `Bearer ${this.apiKey}`,
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({ recipients, message, from }),
    });

    if (!res.ok) {
      const err = await res.json();
      throw new Error(`Tekhook API error: ${err.message || res.statusText}`);
    }

    return res.json();
  }
}

class TekHook {
  constructor(apiKey) {
    if (!apiKey) throw new Error('TEKHOOK_API_KEY is required');
    this.sms = new SMSClient(apiKey);
  }
}

module.exports = { TekHook };

File: tekhook-global-sms-quickstart/python/tekhook.py
# tekhook.py — Lightweight Tekhook API wrapper for Python
# Full API docs: https://tekhook.co

import os
import requests

BASE_URL = "https://api.tekhook.co/v1"


class SMSClient:
    def __init__(self, api_key: str):
        self.api_key = api_key
        self.session = requests.Session()
        self.session.headers.update({
            "Authorization": f"Bearer {api_key}",
            "Content-Type": "application/json",
        })

    def send(self, to: str, message: str, sender: str = "Tekhook") -> dict:
        """Send a single SMS message."""
        response = self.session.post(
            f"{BASE_URL}/sms/send",
            json={"to": to, "message": message, "from": sender},
        )
        response.raise_for_status()
        return response.json()

    def send_bulk(self, recipients: list[str], message: str, sender: str = "Tekhook") -> dict:
        """Send an SMS to multiple recipients."""
        response = self.session.post(
            f"{BASE_URL}/sms/bulk",
            json={"recipients": recipients, "message": message, "from": sender},
        )
        response.raise_for_status()
        return response.json()


class TekHook:
    def __init__(self, api_key: str | None = None):
        api_key = api_key or os.environ.get("TEKHOOK_API_KEY")
        if not api_key:
            raise ValueError("TEKHOOK_API_KEY is required. Set it as an env var or pass it directly.")
        self.sms = SMSClient(api_key)
